pkg/controller/ec2/natgateway/controller.go

```diff
@@ -0,0 +1,204 @@
+		}, nil
+	}
+
+	response, err := e.client.DescribeNatGatewaysRequest(&awsec2.DescribeNatGatewaysInput{
+		NatGatewayIds: []string{meta.GetExternalName(cr)},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ec2.IsNatGatewayNotFoundErr, err), errDescribe)
+	}
+
+	// in a successful response, there should be one and only one object
+	if len(response.NatGateways) != 1 {
+		return managed.ExternalObservation{}, errors.New(errNotSingleItem)
+	}
+
+	observed := response.NatGateways[0]
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errSpecUpdate)
+		}
+	}
+
+	cr.Status.AtProvider = ec2.GenerateNATGatewayObservation(observed)
+
+	switch cr.Status.AtProvider.State {
+	case v1beta1.NatGatewayStatusPending:
+		cr.SetConditions(runtimev1alpha1.Unavailable())
+	case v1beta1.NatGatewayStatusFailed:
+		cr.SetConditions(runtimev1alpha1.Unavailable().WithMessage(aws.StringValue(observed.FailureMessage)))
+	case v1beta1.NatGatewayStatusAvailable:
+		cr.SetConditions(runtimev1alpha1.Available())
+	case v1beta1.NatGatewayStatusDeleting:
+		cr.SetConditions(runtimev1alpha1.Deleting())
+	case v1beta1.NatGatewayStatusDeleted:
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: v1beta1.CompareTags(cr.Spec.ForProvider.Tags, observed.Tags),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1beta1.NATGateway)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+	if err := e.kube.Status().Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errStatusUpdate)
+	}
+
+	nat, err := e.client.CreateNatGatewayRequest(&awsec2.CreateNatGatewayInput{
+		AllocationId: cr.Spec.ForProvider.AllocationID,
+		SubnetId:     cr.Spec.ForProvider.SubnetID,
+		TagSpecifications: []awsec2.TagSpecification{
+			{
+				ResourceType: "natgateway",
+				Tags:         v1beta1.GenerateEC2Tags(cr.Spec.ForProvider.Tags),
+			},
+		},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+
+	meta.SetExternalName(cr, aws.StringValue(nat.NatGateway.NatGatewayId))
+
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errSpecUpdate)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1beta1.NATGateway)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	// This implies additional tags have been manually added to the resource
+	if len(cr.Status.AtProvider.Tags) > len(cr.Spec.ForProvider.Tags) {
```

Comment:
Since `Tags` should not appear under status, I think you'll need to make an AWS call to get the tags.